p2p/node/p2p_services.go

```diff
@@ -77,9 +79,22 @@ func (p *P2PNode) requestFromPeer(peerID peer.ID, location common.Location, data
 	// Check the received data type & hash matches the request
 	switch datatype.(type) {
 	case *types.Block:
-		if block, ok := recvdType.(*types.Block); ok && block.Hash() == data.(common.Hash) {
-			return block, nil
+		if block, ok := recvdType.(*types.Block); ok {
```

Comment:
these bug fixes should go into independent PRs. AFAIK these methods do not exhibit issues in Hussam and Shreekara's testing, so it would be good to know what is being fixed independently of your snap sync PR